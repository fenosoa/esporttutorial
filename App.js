import React, { useState } from 'react';
import {
  AppRegistry,
  StyleSheet,
  Text,
  View,
  TouchableOpacity
} from 'react-native';

const App = () => {
  console.log('data : ' + process.env.TESTAPP)
  const [greeting, setGreeting] = useState(undefined);

  const onButtonPress = (greeting) => {
    setGreeting(greeting);
  };

  const renderAfterButton = () => (
    <View style={{flex: 1, paddingTop: 20, justifyContent: 'center', alignItems: 'center'}}>
      <Text style={{fontSize: 25}}>
        {greeting}!!!
      </Text>
    </View>
  );

  return (
    <>
      <View testID='welcome' style={{flex: 1, paddingTop: 20, justifyContent: 'center', alignItems: 'center'}}>
        <Text style={{fontSize: 25, marginBottom: 30}}>
          Welcome
        </Text>
        <TouchableOpacity testID='hello_button' onPress={() => onButtonPress('Hello')}>
          <Text style={{color: 'blue', marginBottom: 20}}>Say Hello</Text>
        </TouchableOpacity>
        <TouchableOpacity testID='world_button' onPress={() => onButtonPress('World')}>
          <Text style={{color: 'blue', marginBottom: 20}}>Say World</Text>
        </TouchableOpacity>
        <TouchableOpacity testID='goodbye_button' onPress={() => onButtonPress('Goodbye, World')}>
          <Text style={{color: 'blue', marginTop: 50, marginBottom: 20}}>Say Goodbye</Text>
        </TouchableOpacity>
      </View>
      <Text>
       App dotenv: {process.env.TESTAPP}
       try
      </Text>
      {greeting && (renderAfterButton())}
    </>
  );
};

export default App;
