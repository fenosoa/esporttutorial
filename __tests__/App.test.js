/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: import explicitly to use the types shiped with jest.
import {it} from '@jest/globals';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('should render correctly', () => {
  renderer.create(<App />);
});

it('should match the snapshot', () => {
  const snapshot = renderer.create(<App />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
