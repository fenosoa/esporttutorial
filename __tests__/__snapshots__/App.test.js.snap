// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
[
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingTop": 20,
      }
    }
    testID="welcome"
  >
    <Text
      style={
        {
          "fontSize": 25,
          "marginBottom": 30,
        }
      }
    >
      Welcome
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="hello_button"
    >
      <Text
        style={
          {
            "color": "blue",
            "marginBottom": 20,
          }
        }
      >
        Say Hello
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="world_button"
    >
      <Text
        style={
          {
            "color": "blue",
            "marginBottom": 20,
          }
        }
      >
        Say World
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="goodbye_button"
    >
      <Text
        style={
          {
            "color": "blue",
            "marginBottom": 20,
            "marginTop": 50,
          }
        }
      >
        Say Goodbye
      </Text>
    </View>
  </View>,
  <Text>
    App dotenv: 
    test data
    try
  </Text>,
]
`;
